x/params/store/store.go

```diff
@@ -0,0 +1,167 @@
+package store
+
+import (
+	"fmt"
+	"reflect"
+	"regexp"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/cosmos/cosmos-sdk/wire"
+)
+
+// Individual parameter store for each keeper
+type Store struct {
+	cdc  *wire.Codec
+	key  sdk.StoreKey
+	tkey sdk.StoreKey
+
+	space []byte
+}
+
+// NewStore constructs a store with namespace
+func NewStore(cdc *wire.Codec, key sdk.StoreKey, tkey sdk.StoreKey, space string) Store {
+	if !isAlphaNumeric(space) {
+		panic("paramstore space expressions can only contain alphanumeric characters")
+	}
+
+	return Store{
+		cdc:  cdc,
+		key:  key,
+		tkey: tkey,
+
+		space: []byte(space + "/"),
+	}
+}
+
+// Wrapper for key string
+type Key struct {
+	s string
+}
+
+// copied from baseapp/router.go
+var isAlphaNumeric = regexp.MustCompile(`^[a-zA-Z0-9]+$`).MatchString
```

Comment:
TODO don't have to do this now, but we should probably move these into a common place before we end up with a thousand different ones...
There's IsASCIIText() in tendermint/libs/common, maybe it belongs there.